appformer-ala-wildfly-provider/src/main/java/org/kie/appformer/ala/wildfly/executor/AppFormerWildflyRuntimeExecExecutor.java

```diff
@@ -0,0 +1,230 @@
+
+        Path warPath = Paths.get( runtimeConfig.getWarPath( ) );
+        if ( runtimeConfig.getKieDataSource( ) != null || runtimeConfig.getJndiDataSource( ) != null ) {
+
+            boolean hasPersistence = false;
+            try {
+                hasPersistence = provisioningHelper.hasPersistenceSettings( warPath );
+            } catch ( Exception e ) {
+                //un-common case, let the provisioning continue.
+                logger.warn( "It was not possible to establish if the persistence is configured for current application: " + warPath, e );
+            }
+            if ( !hasPersistence ) {
+                logger.warn( "Persistence configuration was not found for current application: " + warPath );
+            } else {
+                if ( isLocal( wildflyProvider ) ) {
+                    localProvisioning( runtimeConfig );
+                } else {
+                    remoteProvisioning( runtimeConfig, wildflyProvider );
+                }
+            }
+        }
+
+        Optional< WildflyRuntime > result = super.apply( ( WildflyRuntimeConfiguration ) runtimeConfig );
+        logger.info( "Executor finished, wildflyRuntime: " + result.get( ) );
+        return result;
+    }
+
+    private void localProvisioning( AppFormerWildflyRuntimeExecConfig runtimeConfig ) throws ProvisioningException {
+        Path warPath = Paths.get( runtimeConfig.getWarPath( ) );
+        String targetDataSource;
+        if ( runtimeConfig.getJndiDataSource( ) != null ) {
+            targetDataSource = runtimeConfig.getJndiDataSource( );
+        } else {
+            try {
+                DataSourceDeploymentInfo deploymentInfo = provisioningHelper.findDataSourceDeploymentInfo( runtimeConfig.getKieDataSource( ) );
+                if ( deploymentInfo == null ) {
+                    throw new ProvisioningException( "Required data source: " + runtimeConfig.getKieDataSource( ) + " is not deployed in current server." );
+                } else {
+                    targetDataSource = deploymentInfo.getJndi( );
+                }
+            } catch ( Exception e ) {
+                final String msg = "It was not possible to get deployment information for the required data source: "
+                        + runtimeConfig.getKieDataSource( );
+                logger.error( msg, e );
+                throw new ProvisioningException( msg, e );
+            }
+        }
+        updatePersistenceSettings( warPath, targetDataSource );
+    }
+
+    private void remoteProvisioning( AppFormerWildflyRuntimeExecConfig runtimeConfig, WildflyProvider wildflyProvider ) {
+        Path warPath = Paths.get( runtimeConfig.getWarPath( ) );
+        String warName = warPath.getFileName().toString();
+
+        WildflyAppState appState = wildfly.getWildflyClient( wildflyProvider ).getAppState( warName );
+        if ( "Running".equals( appState.getState() ) && runtimeConfig.getRedeployStrategy().equals( "auto" ) ) {
```

Comment:
Hello, just a thought - would you consider using constants for the hard-coded values such as "Running" or "auto"?